common/algorithms/simd-int-to-string.cpp

```diff
@@ -183,26 +183,26 @@ char *simd_uint32_to_string(uint32_t value, char *out_buffer) noexcept {
   return out_buffer + 8;
 }
 
-char *simd_int32_to_string(int32_t value, char *out_buffer) noexcept {
+char *simd_int32_to_string(int32_t value, char *out_buffer, size_t buffer_size) noexcept {
   auto u = static_cast<uint32_t>(value);
   if (value < 0) {
     *out_buffer++ = '-';
     u = ~u + 1;
   }
-  return simd_uint32_to_string(u, out_buffer);
+  return simd_uint32_to_string(u, out_buffer, buffer_size);
 }
 
-char *simd_uint64_to_string(uint64_t value, char *out_buffer) {
-  if (value < 10000u) {
+char *simd_uint64_to_string(uint64_t value, char *out_buffer, [[maybe_unused]] size_t buffer_size) {
+  if (value < 10000U) {
     return impl_::uint32_less10000_to_string(static_cast<uint32_t>(value), out_buffer);
   }
-  if (value < 100000000u) {
+  if (value < 100000000U) {
     return impl_::uint32_greater10000_less100000000_to_string(static_cast<uint32_t>(value), out_buffer);
   }
 
-  if (value < 10000000000000000ull) {
-    const auto v0 = static_cast<uint32_t>(value / 100000000u);
-    const auto v1 = static_cast<uint32_t>(value % 100000000u);
+  if (value < 10000000000000000ULL) {
```

Comment:
I'm not sure about tons of such changes. It spoils git history and not easy for review while profit is dubious.
I know clang-tidy highlighted such code, how about to tweak .clang-tidy to just suppress such alert?

Reply:
done